### PR TITLE
fix: prevent stdlib module name conflicts in bundled imports

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

This PR fixes a critical issue where bundled imports could overwrite standard library modules when local submodules share the same name as stdlib modules.

## Problem

When processing relative imports like `from . import abc` where `abc` is both a local submodule and a stdlib module name, the bundler was creating assignments (e.g., `abc = mypkg_abc`) that would overwrite the stdlib `abc` module. This caused runtime errors when code tried to use the stdlib module.

## Solution

Added checks to skip creating assignments that would conflict with standard library modules. The fix is applied in two locations:

1. **In `RecursiveImportTransformer`**: When creating deferred imports for inlined submodules, check if the local name would conflict with a stdlib module
2. **In `create_assignments_for_inlined_imports`**: When creating assignments for imports from inlined modules, skip assignments that would conflict with stdlib modules

## Test plan

- [x] Added comprehensive test fixture `stdlib_module_name_collision` that reproduces the issue
- [x] Verified the test fails without the fix (Python execution error)
- [x] Verified the test passes with the fix (execution succeeds)
- [x] All existing tests continue to pass
- [x] No clippy warnings or errors

## Example

Before fix:
```python
import abc  # stdlib
# ... later in bundled code ...
abc = mypkg_abc  # Overwrites stdlib!
```

After fix:
```python
import abc  # stdlib
# ... later in bundled code ...
# Assignment skipped, stdlib module preserved
local_abc = mypkg_abc  # Only aliased imports are created
```

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: crates/cribo/src/code_generator/import_transformer.rs [hunk 1]
```diff
@@ -2397,6 +2423,21 @@ fn rewrite_import_from(
...
+            if let Stmt::Import(import_stmt) = stmt {
+                for alias in &import_stmt.names {
+                    stdlib_modules.insert(alias.name.as_str().to_string());
+                }
+            }
+        }
+        // Add modules from import-from statements
+        for module_name in bundler.stdlib_import_from_map.keys() {
+            stdlib_modules.insert(module_name.clone());
+        }
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply crates/cribo/src/code_generator/import_transformer.rs]: ✅ Addressed in 56b7a806f. Thanks for the excellent suggestion!

The fix now properly handles all stdlib names in scope including:
- Module aliases: `import os as my_os`
- Imported symbols: `from os import path`  
- Symbol aliases: `from os import path as p`

Created a centralized helper method `collect_stdlib_names_in_scope()` on Bundler to avoid code duplication and ensure consistency across both locations where this check is needed.

File: crates/cribo/src/code_generator/bundler.rs [hunk 2]
```diff
@@ -169,6 +169,42 @@ impl<'a> Bundler<'a> {
...
+                        alias
+                            .asname
+                            .as_ref()
+                            .unwrap_or(&alias.name)
+                            .as_str()
+                            .to_string(),
+                    );
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply crates/cribo/src/code_generator/bundler.rs]: ✅ Addressed in 5670eeb. Thanks for catching this important edge case!

The fix now correctly extracts the top-level module name for dotted imports (e.g., `import importlib.machinery` only binds `importlib`). I also added a test case with a dotted import to verify the behavior.
